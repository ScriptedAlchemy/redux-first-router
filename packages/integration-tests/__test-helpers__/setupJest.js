const config = require('../.testsConfig.json')

// this allows for:
//
// - toggling logging for all tests
// - toggling whether `snap`, `snapPop` and `snapChange` just dispatch or dispatch + snap
//   (useful, for verifying "snipes" work without being cluttered by snapshot diffs)
// - etc
process.env.RUDY_OPTIONS = JSON.stringify(config)
