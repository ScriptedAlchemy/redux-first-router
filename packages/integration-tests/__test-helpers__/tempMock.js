export default (module, fn, reset = true) => {
  if (reset) jest.resetModules()
  jest.doMock(module, fn)
}
