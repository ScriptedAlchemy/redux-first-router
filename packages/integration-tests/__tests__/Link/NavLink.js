import { createNavLink, event } from '../../__test-helpers__/createLink'

test('EXACT (true): DONT show active class', async () => {
  const { tree } = await createNavLink('/second/dog', {
    to: '/second',
    // partial: false, // this is the default
    activeClassName: 'active',
  })
  console.log(tree)
  expect(tree).toMatchSnapshot()
})

test('EXACT (false): SHOW active class', async () => {
  const { tree } = await createNavLink('/second/dog', {
    to: '/second',
    partial: true,
    activeClassName: 'active',
  })
  console.log(tree)
  expect(tree).toMatchSnapshot()
})

test('STRICT: DONT show active class', async () => {
  const { tree } = await createNavLink('/first', {
    to: '/first/',
    strict: true,
    activeClassName: 'active',
  })

  expect(tree).toMatchSnapshot()
})

test('show activeStyle', async () => {
  const { tree } = await createNavLink('/first', {
    to: '/first',
    activeStyle: { color: 'red' },
  })

  expect(tree).toMatchSnapshot()
})

test('combine with existing styles and class', async () => {
  const { tree } = await createNavLink('/first', {
    to: '/first',
    className: 'foo',
    style: { fontSize: 32 },
    activeClassName: 'active',
    activeStyle: { color: 'red' },
  })

  expect(tree).toMatchSnapshot()
})

test('reacts to state changes (onClick)', async () => {
  const { tree, store, component } = await createNavLink('/first', {
    to: '/second/bar',
    activeClassName: 'active',
  })

  expect(tree.props.className).toEqual('')
  expect(tree).toMatchSnapshot() /* ? tree */

  // store.dispatch({ type: 'SECOND', params: { param: 'bar' } })
  await tree.props.onClick(event) // this dispatches above action (obviously)

  const tree2 = component.toJSON() /* ? */

  expect(tree2.props.className).toEqual('active')
  expect(tree2).toMatchSnapshot()
})

test('isActive returns true', async () => {
  const { tree, store } = await createNavLink('/first', {
    to: '/first',
    activeClassName: 'active',
    isActive: (match, location) => {
      expect(match).toMatchSnapshot()
      expect(location).toMatchSnapshot()
      return match
    },
  })

  expect(tree).toMatchSnapshot()
})

test('isActive return false', async () => {
  const { tree, component, store } = await createNavLink('/first', {
    to: '/first',
    activeClassName: 'active',
    isActive: (match, location) => {
      expect(match).toMatchSnapshot()
      expect(location).toMatchSnapshot()
      return location.type === 'SECOND'
    },
  })

  console.log('tree', tree)
  expect(tree).toMatchSnapshot()

  await store.dispatch({ type: 'SECOND', params: { param: 'foo' } })
  expect(component.toJSON()).toMatchSnapshot()
})

it('supports custom HTML tag name', async () => {
  const { tree, store } = await createNavLink('/first', {
    to: '/second',
    activeClassName: 'active',
    component: 'div',
  })

  expect(tree).toMatchSnapshot()
})

it('supports custom HTML tag name in active mode', async () => {
  const { tree, store } = await createNavLink('/first', {
    to: '/first',
    activeClassName: 'active-foo',
    component: 'div',
  })

  expect(tree).toMatchSnapshot()
})

it('supports custom HTML tag name which is still a link', async () => {
  const { tree, store } = await createNavLink('/first', {
    to: 'somewhere',
    component: 'a',
    activeClassName: 'active-foo',
  })
  console.log(tree)
  expect(tree).toMatchSnapshot()
})

test('query params are ommitted', async () => {
  const { tree, store } = await createNavLink('/first', {
    to: '/first?foo=123',
    activeClassName: 'active',
  })
  console.log('STATE', store.getState().location)
  expect(tree).toMatchSnapshot()
})

test.only('paths with spaces', async () => {
  const { tree } = await createNavLink('/second', {
    to: '/second',
    activeClassName: 'active',
  })

  // console.log(tree)
  expect(tree).toMatchSnapshot()
})
