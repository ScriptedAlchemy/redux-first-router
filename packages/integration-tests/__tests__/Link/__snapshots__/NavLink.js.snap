// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EXACT (false): SHOW active class 1`] = `
<a
  aria-current="true"
  className="active"
  href="/second"
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={Object {}}
  target={undefined}
/>
`;

exports[`EXACT (true): DONT show active class 1`] = `
<a
  aria-current={false}
  className=""
  href="/second"
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={undefined}
  target={undefined}
/>
`;

exports[`STRICT: DONT show active class 1`] = `
<a
  aria-current={false}
  className=""
  href="/first/"
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={undefined}
  target={undefined}
/>
`;

exports[`combine with existing styles and class 1`] = `
<a
  aria-current="true"
  className="foo active"
  href="/first"
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={
    Object {
      "color": "red",
      "fontSize": 32,
    }
  }
  target={undefined}
/>
`;

exports[`isActive return false 1`] = `
Object {
  "hash": "",
  "matchedPath": "/first",
  "matchers": Object {
    "hash": undefined,
    "path": "/first",
    "query": undefined,
  },
  "params": Object {},
  "partial": false,
  "query": Object {},
  "state": undefined,
  "type": "FIRST",
}
`;

exports[`isActive return false 2`] = `
Object {
  "basename": "",
  "entries": Array [
    Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/first",
      "search": "",
      "state": Object {},
      "url": "/first",
    },
  ],
  "hasSSR": false,
  "hash": "",
  "index": 0,
  "kind": "load",
  "length": 1,
  "params": Object {},
  "pathname": "/first",
  "prev": Object {
    "basename": "",
    "entries": Array [],
    "hasSSR": false,
    "hash": "",
    "index": -1,
    "kind": "",
    "length": 0,
    "params": Object {},
    "pathname": "",
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "type": "",
    "url": "",
  },
  "query": Object {},
  "scene": "",
  "search": "",
  "state": Object {},
  "status": 200,
  "type": "FIRST",
  "url": "/first",
}
`;

exports[`isActive return false 3`] = `
<a
  aria-current={false}
  className=""
  href="/first"
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={undefined}
  target={undefined}
/>
`;

exports[`isActive return false 4`] = `null`;

exports[`isActive return false 5`] = `
Object {
  "basename": "",
  "entries": Array [
    Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/first",
      "search": "",
      "state": Object {},
      "url": "/first",
    },
    Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/second/foo",
      "search": "",
      "state": Object {},
      "url": "/second/foo",
    },
  ],
  "hasSSR": false,
  "hash": "",
  "index": 1,
  "kind": "push",
  "length": 2,
  "params": Object {
    "param": "foo",
  },
  "pathname": "/second/foo",
  "prev": Object {
    "basename": "",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "key": "345678",
        "pathname": "/first",
        "search": "",
        "state": Object {},
        "url": "/first",
      },
    ],
    "hasSSR": false,
    "hash": "",
    "index": 0,
    "kind": "load",
    "length": 1,
    "params": Object {},
    "pathname": "/first",
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "FIRST",
    "url": "/first",
  },
  "query": Object {},
  "scene": "",
  "search": "",
  "state": Object {},
  "status": 200,
  "type": "SECOND",
  "url": "/second/foo",
}
`;

exports[`isActive return false 6`] = `
<a
  aria-current="true"
  className="active"
  href="/first"
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={Object {}}
  target={undefined}
/>
`;

exports[`isActive returns true 1`] = `
Object {
  "hash": "",
  "matchedPath": "/first",
  "matchers": Object {
    "hash": undefined,
    "path": "/first",
    "query": undefined,
  },
  "params": Object {},
  "partial": false,
  "query": Object {},
  "state": undefined,
  "type": "FIRST",
}
`;

exports[`isActive returns true 2`] = `
Object {
  "basename": "",
  "entries": Array [
    Object {
      "basename": "",
      "hash": "",
      "key": "345678",
      "pathname": "/first",
      "search": "",
      "state": Object {},
      "url": "/first",
    },
  ],
  "hasSSR": false,
  "hash": "",
  "index": 0,
  "kind": "load",
  "length": 1,
  "params": Object {},
  "pathname": "/first",
  "prev": Object {
    "basename": "",
    "entries": Array [],
    "hasSSR": false,
    "hash": "",
    "index": -1,
    "kind": "",
    "length": 0,
    "params": Object {},
    "pathname": "",
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "type": "",
    "url": "",
  },
  "query": Object {},
  "scene": "",
  "search": "",
  "state": Object {},
  "status": 200,
  "type": "FIRST",
  "url": "/first",
}
`;

exports[`isActive returns true 3`] = `
<a
  aria-current="true"
  className="active"
  href="/first"
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={Object {}}
  target={undefined}
/>
`;

exports[`paths with spaces 1`] = `
<a
  aria-current="true"
  className="active"
  href="/second"
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={Object {}}
  target={undefined}
/>
`;

exports[`query params are ommitted 1`] = `
<a
  aria-current="true"
  className="active"
  href="/first?foo=123"
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={Object {}}
  target={undefined}
/>
`;

exports[`reacts to state changes (onClick) 1`] = `
<a
  aria-current={false}
  className=""
  href="/second/bar"
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={undefined}
  target={undefined}
/>
`;

exports[`reacts to state changes (onClick) 2`] = `
<a
  aria-current="true"
  className="active"
  href="/second/bar"
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={Object {}}
  target={undefined}
/>
`;

exports[`show activeStyle 1`] = `
<a
  aria-current="true"
  className="active"
  href="/first"
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={
    Object {
      "color": "red",
    }
  }
  target={undefined}
/>
`;

exports[`supports custom HTML tag name 1`] = `
<div
  aria-current={false}
  className=""
  href={undefined}
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={undefined}
  target={undefined}
/>
`;

exports[`supports custom HTML tag name in active mode 1`] = `
<div
  aria-current="true"
  className="active-foo"
  href={undefined}
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={Object {}}
  target={undefined}
/>
`;

exports[`supports custom HTML tag name which is still a link 1`] = `
<a
  aria-current={false}
  className=""
  href="/somewhere"
  onClick={[Function]}
  onMouseDown={undefined}
  onTouchStart={undefined}
  style={undefined}
  target={undefined}
/>
`;
